### DISCUSSION TITLE
Rename-PnPTenantSite

### DISCUSSION THREAD

**Discussion by Pieter-Veenstra** [pnp/powershell#3449] [General]
I'm not sure where to report this one, as it isn't really a bug, maybe documentation needs to be adjusted or maybe I just don't understand Rename-PnPTenantSite.

I tried out the Rename-PnPTenantSite to update the URL of a site collection within my tenant. This worked well. however while I was testing this Cmdlet I found that as I renamed /sites/examplesite1 to /sites/examplesite2 all worked fine and amazingly I can now access the site through both URLs. I didn't expect that with a Rename.

Then as I did some more testing I tried to rename the site back to /sites/examplesite1 and I got an error that /sites/examplesite1 is already in use.

I'm assuming that rather than renaming the site a redirect is put in place. Is this a temporary thing while the site collection is moved in the background? Or is the a permanent redirect and the rename is really just a redirect? Is there any way (other than migrating content from one site collection to another) to make the original url available again?

**Comment by AndersRask** [comment]
Hi,

this really isn't a PnP "thing", as it is "working as designed" due to the documentation:
The old url remains with a redirect that you must actively delete if you want to rename back
https://learn.microsoft.com/en-us/sharepoint/change-site-address#:~:text=You%20can%27t%20change,redirect.%20Learn%20how

Renaming back
https://learn.microsoft.com/en-us/sharepoint/change-site-address#revert-a-site-address-change

Managing redirects:
https://learn.microsoft.com/en-us/sharepoint/manage-site-redirects

Another thing to be observant about, is the side effects that you may not expect when renaming:
https://learn.microsoft.com/en-us/sharepoint/change-site-address#effects-of-changing-a-site-address


hth
Anders
